adaptivefiltering/execute.py

```diff
@@ -66,14 +66,32 @@ def apply_adaptive_pipeline(
 
     # We decrease the logging level
     logger.setLevel(logging.INFO)
-
     # Ensure existence of output directory
+    # This is not written in the log file
     logger.info(f"Creating output directory {os.path.abspath(output_dir)}")
     os.makedirs(output_dir, exist_ok=True)
 
     # We add a file logging handler
     attach_file_logger(os.path.join(output_dir, "output.log"))
 
+    # if no spatial_refence is defined for the dataset it is tried to be extracted from the metadata
+    if dataset.spatial_reference is None:
+        from adaptivefiltering.pdal import PDALInMemoryDataSet
+
+        dataset.spatial_reference = PDALInMemoryDataSet.convert(
+            dataset
+        ).spatial_reference
```

Comment:
Is there some way of doing this without redundantly converting the dataset?

Reply:
only by forcing the user to manually set the spatial_reference information when loading a dataset.